components/automate-ui/src/app/pages/+compliance/+reporting/+reporting-node/reporting-node.component.ts

```diff
@@ -205,4 +210,44 @@ export class ReportingNodeComponent implements OnInit, OnDestroy {
         this.activeReport = Object.assign(report, data);
       });
   }
+  
+  toggleDownloadDropdown() {
+    this.downloadOptsVisible = !this.downloadOptsVisible;
+  }
+  hideDownloadStatus() {
+    this.downloadStatusVisible = false;
+    this.downloadInProgress = false;
+    this.downloadFailed = false;
+  }
+
+  showDownloadStatus() {
+    this.downloadStatusVisible = true;
+    this.downloadInProgress = true;
+    this.downloadFailed = false;
+  }
+
+  onDownloadNodeReport(format) {
```

Comment:
Always use strong typing.
Let's make this a string literal type by putting this in stats.service.ts:
```
export type reportFormat = 'json' | 'csv';
```
then this line becomes:
```suggestion
  onDownloadNodeReport(format: reportFormat) {
```

Also, use it in stats.service, changing this...
``` 
downloaNodeReport(format: string, reportQuery: ReportQuery): Observable<string> {
```
to this
```
downloadNodeReport(format: reportFormat, reportQuery: ReportQuery): Observable<string> {
```